lib/remote/apilistener-configsync.cpp

```diff
@@ -104,7 +110,32 @@ Value ApiListener::ConfigUpdateObjectAPIHandler(const MessageOrigin::Ptr& origin
 		return Empty;
 	}
 
+	auto isObjectLocked = [&ptype, &objName] {
+		auto type = ApiListener::m_ObjectNames.find(ptype.get());
+		if (type == ApiListener::m_ObjectNames.end()) {
+			return false;
+		}
+
+		return type->second.find(objName) != type->second.end();
+	};
+
+	std::unique_lock<std::mutex> lock(m_ObjectNamesMutex);
+	if (isObjectLocked()) {
```

Comment:
The _if_ shouldn't be necessary.

Reply:
> Any thread that intends to wait on a std::condition_variable must:
> 1. Acquire a [std::unique_lock](http://en.cppreference.com/w/cpp/thread/unique_lock)<[std::mutex](http://en.cppreference.com/w/cpp/thread/mutex)> on the mutex used to protect the shared variable.
> 2. Do one of the following:
>    1. Check the condition, in case it was already updated and notified.

[`cppreference`](https://en.cppreference.com/w/cpp/thread/condition_variable) clearly says to do so! Why would you want to enter a thread in a wait mode that doesn't have to?

Reply:
And it says that your wait() is equivalent to:

```c++
while (!!isLocked())
{
    wait(lock);
}
```

Reply:
You didn't quote the "or" branch of the "Do one of the following":

> Use the predicated overload of `wait`, `wait_for`, and `wait_until`, which performs the same three steps.